calendar-amqp/src/main/java/com/linagora/calendar/amqp/EventITIPConsumer.java

```diff
@@ -0,0 +1,139 @@
+import reactor.rabbitmq.ConsumeOptions;
+import reactor.rabbitmq.ExchangeSpecification;
+import reactor.rabbitmq.QueueSpecification;
+import reactor.rabbitmq.Receiver;
+import reactor.rabbitmq.Sender;
+
+public class EventITIPConsumer implements Closeable, Startable {
+    public static final String EXCHANGE_NAME = "calendar:itip:deliver";
+    public static final String QUEUE_NAME = "tcalendar:itip:deliver";
+    public static final String DEAD_LETTER_QUEUE = "tcalendar:itip:deliver:dead-letter";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventITIPConsumer.class);
+    private static final boolean REQUEUE_ON_NACK = true;
+
+    private final ReceiverProvider receiverProvider;
+    private final CalDavClient calDavClient;
+
+    private Disposable consumeDisposable;
+
+    @Inject
+    public EventITIPConsumer(ReactorRabbitMQChannelPool channelPool,
+                             @Named(INJECT_KEY_DAV) Supplier<QueueArguments.Builder> queueArgumentSupplier,
+                             CalDavClient calDavClient) {
+        this.receiverProvider = channelPool::createReceiver;
+        this.calDavClient = calDavClient;
+
+        Sender sender = channelPool.getSender();
+        Flux.concat(
+                sender.declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME)
+                    .durable(DURABLE).type(BuiltinExchangeType.FANOUT.getType())),
+                sender.declareQueue(QueueSpecification
+                    .queue(DEAD_LETTER_QUEUE)
+                    .durable(DURABLE)
+                    .arguments(queueArgumentSupplier.get().build())),
+                sender.declareQueue(QueueSpecification
+                    .queue(QUEUE_NAME)
+                    .durable(DURABLE)
+                    .arguments(queueArgumentSupplier.get().deadLetter(DEAD_LETTER_QUEUE).build())),
+                sender.bind(BindingSpecification.binding()
+                    .exchange(EXCHANGE_NAME)
+                    .queue(QUEUE_NAME)
+                    .routingKey(EMPTY_ROUTING_KEY)))
+            .then()
+            .block();
+    }
+
+    public void init() {
+        start();
+    }
+
+    public void start() {
+        consumeDisposable = doConsumeMessages();
+    }
+
+    public void restart() {
+        close();
+        start();
+    }
+
+    @Override
+    public void close() {
+        if (consumeDisposable != null && !consumeDisposable.isDisposed()) {
+            consumeDisposable.dispose();
+        }
+    }
+
+    private Disposable doConsumeMessages() {
+        return delivery(QUEUE_NAME)
+            .flatMap(this::consumeMessage, DEFAULT_CONCURRENCY)
+            .subscribeOn(Schedulers.boundedElastic())
+            .subscribe();
+    }
+
+    public Flux<AcknowledgableDelivery> delivery(String queue) {
+        return Flux.using(receiverProvider::createReceiver,
+            receiver -> receiver.consumeManualAck(queue, new ConsumeOptions().qos(DEFAULT_CONCURRENCY)),
```

Comment:
QOS and flatMap concurrency need to be configurable (at least using a JVM property) in my opinion.